apps/studio/src/components/PageEditor/LinkEditorModal.tsx

```diff
@@ -211,34 +218,29 @@ export const LinkEditorModal = ({
   </Modal>
 )
 
-const ModalLinkEditor = ({
-  onChange,
-}: {
-  onChange: (value: string) => void
-}) => {
+const ModalLinkEditor = () => {
   const { error, curHref, setHref } = useLinkEditor()
   const { siteId } = useQueryParse(editSiteSchema)
-  const handleChange = (value: string) => {
-    onChange(value)
-    setHref(value)
-  }
 
   return (
     <LinkHrefEditor
       label="Link destination"
       description="When this is clicked, open:"
       isRequired
       isInvalid={!!error}
-      pageLinkElement={
-        <PageLinkElement value={curHref} onChange={handleChange} />
-      }
+      pageLinkElement={<PageLinkElement value={curHref} onChange={setHref} />}
       fileLinkElement={
-        <FileAttachment
-          maxSizeInBytes={MAX_FILE_SIZE_BYTES}
-          acceptedFileTypes={FILE_UPLOAD_ACCEPTED_MIME_TYPE_MAPPING}
-          siteId={siteId}
-          setHref={(href) => handleChange(href ?? "")}
-        />
+        getLinkHrefType(curHref) === LINK_TYPES.File ? (
+          <AttachmentData data={curHref} onClick={() => setHref("")} />
+        ) : (
+          <FileAttachment
+            maxSizeInBytes={MAX_FILE_SIZE_BYTES}
+            acceptedFileTypes={FILE_UPLOAD_ACCEPTED_MIME_TYPE_MAPPING}
+            siteId={siteId}
+            setHref={(href) => setHref(href ?? "")}
+            shouldFetchResource={false}
+          />
+        )
```

Comment:
no need to change but thought to point out that naming does feel a bit counterintuitive, that we use `FileAttachment` when link types isn't File

Reply:
yea it's not ideal - do you have any suggestions? cos this covers basically everything we wanna upload so `File` felt like the most general term. wasn't specifically referring to `LINK_TYPES` here